### DISCUSSION TITLE
Adding language support

### DISCUSSION THREAD

**Discussion by herrwinfried** [AvaloniaUI/Avalonia#10061] [Q&A]
I want to add language support to my project. I did research for this, I looked at the codes, but it didn't help.

**Comment by herrwinfried** [comment]
Oh, I'm sorry. I forgot to write the required code in the app.axaml. I would like to mention it a little, as I think that others may look at it in the future.

https://github.com/timunie/Tims.Avalonia.Samples/tree/36e9e99f0740b533161d2a6508b190f6d8637360/src/LocalizationSample

I checked the project in this link. Then I realized that there are codes that I need to add to my project.

I reviewed [App.axaml](https://github.com/timunie/Tims.Avalonia.Samples/blob/36e9e99f0740b533161d2a6508b190f6d8637360/src/LocalizationSample/App.axaml#L9L16), [ViewModels/MainWindowViewModel.cs](https://github.com/timunie/Tims.Avalonia.Samples/blob/36e9e99f0740b533161d2a6508b190f6d8637360/src/LocalizationSample/ViewModels/MainWindowViewModel.cs#L14) codes and added the necessary parts to my project.

I then visited [Assets/Lang/](https://github.com/timunie/Tims.Avalonia.Samples/tree/36e9e99f0740b533161d2a6508b190f6d8637360/src/LocalizationSample/Assets/Lang) for a sample lang.

I took a look at [Views/MainWindow.axaml](https://github.com/timunie/Tims.Avalonia.Samples/blob/36e9e99f0740b533161d2a6508b190f6d8637360/src/LocalizationSample/Views/MainWindow.axaml) to understand how they use the code/command.

**Comment by herrwinfried** [comment]
But I can't use [MainWindow.axaml](https://github.com/timunie/Tims.Avalonia.Samples/blob/36e9e99f0740b533161d2a6508b190f6d8637360/src/LocalizationSample/Views/MainWindow.axaml#L12.14)
Design.DataContext gives an error when you place it as in the code.
My code starts like this.


```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:AvaloniaApplication.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:views="clr-namespace:AvaloniaApplication.Views"
        x:Class="AvaloniaApplication.Views.MainWindow"
        Title="[BETA] Hello world"
        WindowStartupLocation="CenterScreen"
        Height="530"
        Width="1200"
        CanResize="False">

    <Canvas>
        
        <DockPanel>
```

**Reply by timunie** [reply]
How did you create your new project? Avalonia has some templates for new projects and also to add windows and user controls etc. Make sure you have the extensions installed in your IDE.

https://docs.avaloniaui.net/docs/getting-started/ide-support

**Reply by herrwinfried** [reply]
> How did you create your new project? Avalonia has some templates for new projects and also to add windows and user controls etc. Make sure you have the extensions installed in your IDE.
> 
> https://docs.avaloniaui.net/docs/getting-started/ide-support

I have successfully installed Jetbrains Rider. I chose your structure which is Cross-Platform Application.
